src/App.tsx

```diff
@@ -15,48 +17,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  length = 'Sort by length',
+  Alphabetically = 'Sort alphabetically',
+  clear = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  isReverse: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((goodA, goodB) => {
+      switch (sortField) {
+        case SortType.Alphabetically:
+          return goodA.localeCompare(goodB);
+        case SortType.length:
+          return goodA.length - goodB.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.clear);
+  const [isReverse, setIsReverse] = useState(false);
+
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    sortField,
+    isReverse,
+  );
+
+  const handleReverse = () => setIsReverse(!isReverse);
```

Comment:
I believe this halper is redundant,  you can call setIsReverse directly onClick.
Also when you set new state based on the previous one always use callback:

```typescript
setIsReverse(curr => !curr);
```